packages/pluggableWidgets/barcode-scanner-web/src/hooks/useCustomErrorMessage.ts

```diff
@@ -0,0 +1,33 @@
+import { Exception } from "@zxing/library";
+import { useState, useCallback } from "react";
+
+type ErrorCb<E extends Error = Error> = (error: E) => void;
+
+type UseCustomErrorMessageHook = () => [string | null, ErrorCb];
+
+function getErrorMessage<E extends Error>(error: E): string | null {
+    // Error comes from `zxing`
+    if (error instanceof Exception) {
+        return "Error in barcode scanner: an unexpected error occurred while detecting a barcode in the camera media stream.";
+    }
+
+    if (error.name === "NotFoundError") {
+        return "Error in barcode scanner: no camera media devices were found.";
+    }
+
+    // Ignore error if user restricted camera access
+    if (error.name === "NotAllowedError") {
+        return null;
```

Comment:
is it notifying user somehow that we need access? maybe on device level?

Reply:
Yes, this happens on browser level, we just ignoring if user denies access